02-microarray/pathway-analysis_microarray_02_ora_with_webgestaltr.Rmd

```diff
@@ -0,0 +1,260 @@
+
+The `WebGestaltR` function is a wrapper for the [WEB-based GEne SeT AnaLysis Toolkit (WebGestalt)](http://www.webgestalt.org/) [@Wang2017].
+
+WebGestalt can perform multiple _types_ of pathway analysis. 
+Here we're using it for ORA and we will use [Gene Ontology (GO)](http://geneontology.org/docs/ontology-documentation/) biological processes as our source of gene sets.
+We can see all supported gene sets for humans with `listGeneSet()` function.
+We are using `head()` so we just get a preview and not the whole table printed out. 
+
+```{r}
+head(listGeneSet("hsapiens"))
+```
+
+The `WebGestaltR()` function will generate an HTML file with a report (path specified by the `outputDirectory` and `projectName` arguments) _and_ return a data frame.
+
+```{r}
+go_enrichment_results <-
+  WebGestaltR(
+    enrichMethod = "ORA",
+    organism = "hsapiens",
+    enrichDatabase = "geneontology_Biological_Process", # What categories we'd like enrichment results for
+    interestGene = upregulated_genes, # The vector of our genes of interest
+    interestGeneType = "ensembl_gene_id",
+    referenceGene = all_genes, # The vector of all the genes originally in our set
+    referenceGeneType = "ensembl_gene_id",
+    outputDirectory = results_dir, # Replace with the directory you would like your results saved to
+    projectName = "GSE37418_up_SHH_lfc2_fdr0_05" # Replace with a relevant output name for the HTML file
+  )
+```
+
+As noted in the messages from `WebGestaltR`, the results are in `results/Project_GSE37418_up_SHH_lfc2_fdr0_05`.
+The HTML report is at [`results/Project_GSE37418_up_SHH_lfc2_fdr0_05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html`](./results/Project_GSE37418_up_SHH_lfc2_fdr0_05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html).
+
+## Read in a results table
+
+Let's take a look at one of the results tables look like by reading it into R. 
+It is a `.txt` file, however its values are separated by tabs, making it a TSV; hence we can read it in with `readr::read_tsv()` function. 
+
+```{r}
+results_df <- readr::read_tsv(
+  file.path(
+    results_dir,
+    "Project_GSE37418_up_SHH_lfc2_fdr0_05", # This will need to change depending on what you supply to `projectName` argument in `WebGestaltR()` function
+    "enrichment_results_GSE37418_up_SHH_lfc2_fdr0_05.txt" # Same here, but the prefix is different
+  )
+)
+```
+
+Let's print out a preview of it.   
+
+```{r}
+head(as.data.frame(results_df)) # We are coercing to a data frame (from a tibble) so all results are rendered a bit more clearly
+```
+It looks like some of the top results are pathways associated with ribosomes and translation.
+Gene sets, particularly GO gene sets that are of a hierarchical nature, are not independent, so it's important to keep in mind that many of these gene sets could be telling us the same thing.
```

Comment:
This is part of the reason we don't cover GO in training... can you remind me if KEGG is supported by `WebGestaltR`? KEGG pathways overlap, but there's no hierarchy.

Reply:
There is KEGG as well as 63 other options, most of which are based on TCGA network analyses from what I'm understanding. I think KEGG is the option that seems most widely applicable.